### ISSUE TITLE
jep 4.0.1 disappeared from conda-forge suddenly

### ISSUE THREAD

**Issue by SemanticBeeng** [ninia/jep#387]
**Describe the problem**

Creating `conda` environments with `jep` 4.0.1 as dependency (not `pip`).
Worked fine and then stopped working.
More precisely it fails to resolve (https://anaconda.org/conda-forge/jep used to show 4.0.1 as latest version) and then conda tries a build and that fails. 

Noticed the latest is now 4.0.3.
How is the deployment to conda-forge supposed to work?
Please advise with details on the build and release to better make sense of why this happens.

**Environment (please complete the following information):**
- OS Platform, Distribution, and Version: Ubuntu 20.04 - 20.10
- Python Distribution and Version: 3.8
- Java Distribution and Version:
- Jep Version: 4.0.1
- gcc: 10.3.0
- Python packages used (e.g. numpy, pandas, tensorflow):

**Logs**
![image](https://user-images.githubusercontent.com/9296536/160889044-fb08b530-4db6-4f04-87d9-12230b28039e.png)

**Additional context**

**Comment by SemanticBeeng** [comment]
Hmm ... okay.
For the record, the build is done with this https://github.com/conda-forge/jep-feedstock.

**Comment by ndjensen** [comment]
Interesting, looks like it doesn't like the static assert line in jep_platform.h.  https://github.com/ninia/jep/blob/v4.0.3/src/main/c/Include/jep_platform.h#L38  I don't understand why.

**Comment by SemanticBeeng** [comment]
Indeed.
Unsure what controls the `PY_MAJOR_VERSION` variable.
Only found two references https://github.com/search?q=org%3Aconda-forge+PY_MAJOR_VERSION&type=issues

**Comment by bsteffensmeier** [comment]
`Py_MAJOR_VERSION` is brought in from [patchlevel.h](https://github.com/python/cpython/blob/v3.10.4/Include/patchlevel.h#L19) which is the first thing [Python.h](https://github.com/python/cpython/blob/v3.10.4/Include/Python.h) brings in. I would be surprised if conda-forge is doing anything special with this as it is standard in python builds and we have always had various checks against it.

**Comment by bsteffensmeier** [comment]
I did some investigation and found the conda_forge environemnt for older versions of python does not define `static_assert` in `assert.h`. I've added an extra check in d8cd41bfd1 so we won't use static_assert unless we need to. This will resolve that particular error but I do not have any way to test the conda forge build so if anyone knows how to do that it would be helpful to make sure there aren't other errors.
